lib/detect-port.js

```diff
@@ -8,26 +8,40 @@ module.exports = (port, callback) => {
     callback = port;
     port = null;
   }
+  port = parseInt(port) || 0;
+  let maxPort = port + 10;
+  if (maxPort > 65535) {
+    maxPort = 65535;
+  }
+  debug('detect free port between [%s, %s)', port, maxPort);
   if (typeof callback === 'function') {
-    return tryListen(port, callback);
+    return tryListen(port, maxPort, callback);
   }
   // promise
   return new Promise(resolve => {
-    tryListen(port, (_, realPort) => {
+    tryListen(port, maxPort, (_, realPort) => {
       resolve(realPort);
     });
   });
 };
 
-function tryListen(port, callback) {
-  port = parseInt(port) || 0;
+function tryListen(port, maxPort, callback) {
   const server = new net.Server();
 
   server.on('error', err => {
     debug('listen %s error: %s', port, err);
-    port = 0;
+    if (port === 0) {
+      return callback(err);
```

Comment:
random port won't error on normal case. If listen random port error happen, it's hard to try again, maybe "file handle" full on that time.